packages/react-native-reanimated/src/core.ts

```diff
@@ -33,8 +33,20 @@ export {
 
 const EDGE_TO_EDGE = isEdgeToEdge();
 
-/** @returns `true` in Reanimated 3, doesn't exist in Reanimated 2 or 1 */
-export const isReanimated3 = () => true;
+/**
+ * @deprecated Please use the exported variable `reanimatedMajorVersion`
+ *   instead.
+ * @returns `true` in Reanimated 3, doesn't exist in Reanimated 2 or 1
+ */
+export const isReanimated3 = () => {
+  logger.warn(
+    'The `isReanimated3` function is deprecated. Please use the exported variable `reanimatedMajorVersion` instead.'
+  );
+  return false;
+};
+
+/** @returns Reanimated major version, doesn't exist before Reanimated 4 */
+export const reanimatedMajorVersion = 4;
```

Comment:
We can use value from here - https://github.com/software-mansion/react-native-reanimated/blob/main/packages/react-native-reanimated/src/platform-specific/jsVersion.ts